src/hooks/useFetchImages.ts

```diff
@@ -1,71 +1,78 @@
 import { useEffect, useState } from 'react'
+import { useQuery } from '@tanstack/react-query'
 import unixToDate from '../utilities/date-lib/unixToDate'
 import dateToUnix from '../utilities/date-lib/dateToUnix'
 import unixNow from '../utilities/date-lib/unixNow'
 import ky from 'ky'
 
-const useFetchImages = (needMoreImages: boolean) => {
-  const [isLoading, setIsLoading] = useState(false)
-  const [images, setImages] = useState<Image[]>([])
+type UseFetchImagesResult = {
+  isLoading: boolean
+  error?: Error
+  data: Image[]
+}
+
+const useFetchImages = (needMoreImages: boolean): UseFetchImagesResult => {
   const [increment, setIncrement] = useState(1)
   const [startDate, setStartDate] = useState(unixToDate(unixNow - 518400)) // start w/ last 7 days of data
   const [endDate, setEndDate] = useState(unixToDate(unixNow))
 
-  // Updating time frames once week of images has been called
-  useEffect(() => {
-    setEndDate(() => {
-      const startDateAsUnix = dateToUnix(startDate)
-      const dayBeforePrevStart = unixToDate(startDateAsUnix - 43200)
-
-      return dayBeforePrevStart
-    })
-
-    setStartDate((prevStart) => {
-      const prevStartAsUnix = dateToUnix(prevStart)
-      const sevenDaysFromStart = unixToDate(prevStartAsUnix - 518400)
-
-      return sevenDaysFromStart
-    })
-  }, [increment, isLoading])
-
-  // Api Call
-  useEffect(() => {
-    if (!needMoreImages) return
-
-    const requestApiData = async () => {
+  const {
+    isLoading,
+    error,
+    data: images,
+  } = useQuery({
+    enabled: needMoreImages,
+    queryKey: ['images'],
+    queryFn: async () => {
       setIncrement((count) => (count += 1))
-      setIsLoading(true)
-
-      // @ts-expect-error
-      const env = import.meta.env.VITE_NASA_API_KEY
-      const URL = `https://api.nasa.gov/planetary/apod` // ?api_key=${env}&start_date=${startDate}&end_date=${endDate}
 
       try {
+        // @ts-expect-error
+        const ENV = import.meta.env.VITE_NASA_API_KEY
+        const URL = `https://api.nasa.gov/planetary/apod`
         const request = await ky.get(URL, {
           retry: 3,
           searchParams: {
-            api_key: env,
+            api_key: ENV,
             start_date: startDate,
             end_date: endDate,
           },
         })
+
         if (request.status === 200) {
-          // reversing so today's image shows first
-          const data: Image[] = await request.json()
-          data.reverse()
+          const nasaData: Image[] = await request.json()
 
-          setImages((prevImages) => [...prevImages, ...data])
-          setIsLoading(false)
-        } else console.error('Failed with status code: ', request.status, request.statusText)
+          return nasaData.reverse() // reversing so today's image shows first
+        } else {
+          console.error('Failed with status code: ', request.status, request.statusText)
```

Comment:
Now, useQuery will expect a result of images, everywhere the query fails, you could send an empty array
```suggestion
          console.error('Failed with status code: ', request.status, request.statusText)
          return []
```